src/plugins/presentation_panel/public/panel_component/panel_header/presentation_panel_hover_actions.tsx

```diff
@@ -0,0 +1,603 @@
+      iconType="boxesVertical"
+    />
+  );
+
+  const dragHandle = (
+    <EuiIcon
+      type="move"
+      color="text"
+      className={`${viewMode === 'edit' ? 'embPanel--dragHandle' : ''}`}
+      aria-label={i18n.translate('presentationPanel.dragHandle', {
+        defaultMessage: 'Move panel',
+      })}
+      data-test-subj="embeddablePanelDragHandle"
+      css={css(`
+        margin: ${euiThemeVars.euiSizeXS}; 
+        cursor: move !important;
+        img {
+          pointer-events: all !important;
+        }`)}
+    />
+  );
+
+  return (
+    <div
+      onMouseOver={updateCombineHoverActions}
+      onFocus={updateCombineHoverActions}
+      ref={anchorRef}
+      className="embPanel__hoverActionsAnchor"
+      data-test-embeddable-id={api?.uuid}
+      data-test-subj={`embeddablePanelHoverActions-${(title || defaultTitle || '').replace(
+        /\s/g,
+        ''
+      )}`}
+      css={css({
+        position: 'relative',
+        height: '100%',
+      })}
+    >
+      {children}
+      {api ? (
+        <div
+          ref={hoverActionsRef}
+          css={css({
+            height: euiThemeVars.euiSizeXL,
+            position: 'absolute',
+            display: 'flex',
+            justifyContent: 'space-between',
+            padding: `0 ${euiThemeVars.euiSize}`,
+            flexWrap: 'nowrap',
+            minWidth: '100%',
+            // Prevent hover actions wrapper from blocking interactions with other panels
+            pointerEvents: 'none',
+          })}
+          className={classNames('embPanel__hoverActionsWrapper', {
+            'embPanel__hoverActionsWrapper--openContextMenu': isContextMenuOpen,
+          })}
+        >
+          {viewMode === 'edit' && !combineHoverActions ? (
+            <div
+              ref={leftHoverActionsRef}
+              data-test-subj="embPanel__hoverActions__left"
+              className={classNames(
+                'embPanel__hoverActions',
+                'embPanel__hoverActionsLeft',
+                className
+              )}
+              css={css({ ...borderStyles, ...HOVER_ACTION_STYLES })}
+            >
+              {dragHandle}
+            </div>
+          ) : (
+            <div /> // necessary for the right hover actions to align correctly when left hover actions are not present
+          )}
+          <div
+            ref={rightHoverActionsRef}
+            data-test-subj="embPanel__hoverActions__right"
+            className={classNames(
+              'embPanel__hoverActions',
+              'embPanel__hoverActionsRight',
+              className
+            )}
+            css={css({ ...borderStyles, ...HOVER_ACTION_STYLES })}
+          >
+            {menuPanelsLoading ? (
+              <>
+                <EuiSkeletonRectangle width="16px" height="16px" />
+                <EuiSkeletonRectangle width="16px" height="16px" />
+                <EuiSkeletonRectangle width="16px" height="16px" />
```

Comment:
I wonder if this is better?

```suggestion
                <EuiSkeletonRectangle width="48px" height="16px" />
```

| Current | Suggested |
|--------|--------|
| ![Sep-18-2024 16-06-47](https://github.com/user-attachments/assets/dcab4f71-8834-4e1d-b086-496d498020dc) | ![Sep-18-2024 16-10-41](https://github.com/user-attachments/assets/e1a38393-5b55-469b-9396-6087425477e4) | 

We could also use `EuiFlexGroup` with an `xs` gutter, but this would mean that the loading state is wider than when the actions load (since the icons **don't** have padding once loaded in) 🤔

Reply:
<img width="297" alt="Screenshot 2024-09-30 at 10 32 31 PM" src="https://github.com/user-attachments/assets/62c21866-d5a2-48de-84ff-40f505fa4dc2">

I think adding the gutter for the skeletons is a nice touch even if it doesn't exactly match the width of the resulting hover actions. I made the size 24px instead of 16 to better match the actual size of the hover action icon buttons. I also opted to make it two icons because at a minimum, there should always be at least 2 buttons for every panel.

<img width="168" alt="Screenshot 2024-09-30 at 10 32 44 PM" src="https://github.com/user-attachments/assets/d3425656-3084-415f-8fba-dfd5a3e6fd78">

Reply:
Love it :+1: